pyximport/test/test_pyximport.py

```diff
@@ -68,5 +77,27 @@ def make_ext(name, filename):
     remove_tempdir(tempdir)
 
 
-if __name__=="__main__":
-    test()
+def test_zip():
+    try:
+        import test_zip_module
+    except ImportError:
+        pass
+    else:
+        assert False, "test_zip_module already exists"
+
+    fd, zip_path = tempfile.mkstemp(suffix=".zip")
+    os.close(fd)
+    try:
+        with ZipFile(zip_path, "w") as zf:
+            zf.writestr("test_zip_module.pyx", b"x = 42")
+
+        sys.path.insert(0, zip_path)
+        import test_zip_module
+        assert test_zip_module.x == 42
+    finally:
+        os.remove(zip_path)
```

Comment:
I'd be explicit about removing the exact path that was added. The import executes a lot of code, there is not guarantee that `sys.path` still looks the same afterwards.